### ISSUE TITLE
Add support for gutenberg registration support parameters

### ISSUE THREAD

**Issue by JasonTheAdams** [CMB2/CMB2#1191]
With Gutenberg around the corner, it would be great to be able to take advantage of the custom parameters introduced with Gutenberg: https://wordpress.org/gutenberg/handbook/extensibility/meta-box/

This would allow devs to mark metaboxes as incompatible, compatible, or a fallback.

If I have time at some point in the next month or so I'll make a PR, but it shouldn't be much work so if someone else gets to the faster, great!

**Comment by JiveDig** [comment]
Not sure where to post this note... but we're currently experience issues where things blow up (the entire metabox doesn't load) if you are using a WYSIWYG (TinyMCE) field. See https://github.com/WordPress/gutenberg/issues/7176 for reference.

**Comment by JiveDig** [comment]
Sorry for hijacking... just realize this may have been dealt with here? https://github.com/CMB2/CMB2/issues/1156

**Comment by tw2113** [comment]
@JiveDig Yes I believe that's the issue that was related to other GB conflicts. Worth trying Justin's changes to help test/confirm.

**Comment by JasonTheAdams** [comment]
@jtsternberg Wow, that was the fastest issue close I've ever seen! Thanks! 😄
